### DISCUSSION TITLE
Lettuce 6.x throws commandTimeoutException

### DISCUSSION THREAD

**Discussion by shadymohs** [redis/lettuce#2167] [Q&A]
Dear Sir/Madam,

I used to connect to Redis using spring boot 2.0.5.RELEASE that has dependency for lettuce 5.0.5.RELEASE, and everything was working fine. I upgraded my spring boot to 2.4.13 that has dependency for letttuce 6.0.8.RELEASE. Unfortunatly after upgrade I can't connect to redis anymore due to commandTimeoutException.

Redis configuration class is using autocinfiguration for lettuceConnectionFactory from application.properities: spring.redis.host= spring.redis.port= spring.redis.database= spring.redis.timeout=5s then I create redisTemplate bean

I created a simple java console project and used different lettuce 6.x to 6.2.0 versions, but unfortunatlly no way to solve the issue.

```java
public static void main(String[] args) {
  RedisClient redisClient = RedisClient.create("redis://server-ip:my-port/0");
// no auth so I didn't add password@ RedisClient.create("redis://password@server-ip:my-port/0");

  StatefulRedisConnection<String, String> connection redisClient.connect();
  System.out.println("Connected to Redis");
  connection.close();
  redisClient.shutdown();
}
```
Please for your kind help and support.

Our redis version is standalone 6.0.4, also no auth usage so I don't add password or user name

**Comment by shadymohs** [comment]
I added code for both sample project using main method and how I configure in spring boot

**Reply by yangty89** [reply]
Maybe you can try the code below to set a longer timeout:
```
redisClient.setOptions(ClientOptions.builder()
        .timeoutOptions(TimeoutOptions.builder().fixedTimeout(Duration.ofSeconds(1)).build()).build());
```

**Reply by shadymohs** [reply]
Thanks a lot for your support.
Unfortunatelly I conffigured timeout up to an hour and issue still exist.
Is it required in lettuce 6.x to use password in redisClient Create as below?
RedisClient.create("redis://password@server-ip:my-port/0");

**Reply by tishun** [reply]
> Is it required in lettuce 6.x to use password in redisClient Create as below?

No, not required, even if it were, it should not result in connection timed out

**Comment by tishun** [comment]
We would not be able to help without a way to reproduce this issue.
If you have a reproducer please provide it in #2168 and reopen it
